### ISSUE TITLE
Dialog polyfill is missing role="dialog"

### ISSUE THREAD

**Issue by marcysutton** [GoogleChrome/dialog-polyfill#63]
At this time, only Chrome Canary exposes the dialog role for the HTML5 element. The polyfill should add `role="dialog"` to improve support for assistive technologies in all browsers. Here is a demo I'm using to check semantics and focus management in various browsers: http://s.codepen.io/marcysutton/full/XbyReE 

And here is some more info about the role: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_dialog_role

**Comment by lozandier** [comment]
@marcysutton No longer busy w/ political campaign stuff, I decided to finally address this. Let me know if you find your issue is still not resolved as much as you would've liked it to with my PR.

I also want to address "the last focus element should be focused" problem that even the native dialog has by leveraging merely saving what was referenced in `document.activeElement` or perhaps something that is saved after `dialog.showModal` but I think that would be blocked at the moment. 

I'd appreciate your help clearing that up besides what I've interpreted from the WhatWG spec about the element.
